### DISCUSSION TITLE
convert prisma to pocketbase

### DISCUSSION THREAD

**Discussion by sphade** [pocketbase/pocketbase#4237] [Q&A]
can someone help me convert tis prisma code to pocketbase, all relations expansion are back relations, except for categories, an having issues with the purchase part
```
 const courses = await db.courses.findMany({
	where: {
		isPublished: true,
		title: {
			contains:title,
		},
		categoryId,
	},
	include: {
		category: true,
		chapters: {
			where: {
				isPublished:true,
			},
			select: {
				id: true,
				
			}
		},
		purchase: {
			where: {
				userId,
			}
		}
	},
	orderBy: {
		createdAt:"desc"
	}
}),
```
 this was my implementation, but it a little off 
```
const courses = await pb
			.collection('courses')
			.getFullList<Course>({
				filter: `isPublished = ${true} && title ?~ "${title}"  && category = "${categoryId}" chapters(course).isPublished = ${true} && purchase(course).user = "${userId}`,
				expand: 'category,purchase(course),chapters(course)',
				fields: '*,expand.*,expand.chapters(course).id',
				sort: '-created'
			}) 
```

**Comment by ganigeorgiev** [comment]
Indirect relation filtering is not supported at the moment, aka. the statements `chapters(course).isPublished = ${true}` and `purchase(course).user = "${userId}` are invalid. I'm also not sure what is actually the desired behavior in this case because `filter` will only filter the main records, aka. the "courses", not the expand results; if you want to restrict what is being expanded you can adjust the View API rules of the specific collections (or just send separate list request(s) to the expanded collection(s)).
